File: api/src/models/bid.py
from sqlalchemy import Column, String, Integer, ForeignKey
from sqlalchemy.orm import relationship
from models.base import Base, Entity

class Bid(Base, Entity):
    __tablename__ = 'bids'

    quote_uuid = Column(String, ForeignKey('quotes.uuid'), nullable=False)
    service_uuid = Column(String, ForeignKey('services.uuid'), nullable=False)
    amount = Column(Integer, nullable=False)
    stripe_price_id = Column(String, nullable=True)

    quote = relationship('Quote', back_populates='bids')
    service = relationship('Service', back_populates='bids')

File: frontend/src/models/Bid.js
import Service from './Service';

export default class Bid {
    constructor({
        uuid = '',
        service_uuid = '',
        service = null,
        amount = 0,
        stripe_price_id = ''
    }) {
        this.uuid = uuid;
        this.service_uuid = service_uuid;
        this.service = new Service(service);
        this.amount = amount / 100;
        this.stripe_price_id = stripe_price_id;
    }

    serialize() {
        return {
            amount: parseInt(this.amount*100),
            service_uuid: this.service_uuid,
        };
    }
}